Cannot build due to gonum.org timeout

**What steps did you take and what happened:**
Run `make` or any other make target that ends up building the `velero` binary.

Receive the following output:

```
make[1]: Entering directory '/home/nrb/go/src/github.com/vmware-tanzu/velero'
building: _output/bin/linux/amd64/velero
make shell CMD="-c '\
        GOOS=linux \
        GOARCH=amd64 \
        VERSION=master \
        PKG=github.com/vmware-tanzu/velero \
        BIN=velero \
        OUTPUT_DIR=/output/linux/amd64 \
        ./hack/build.sh'"
make[2]: Entering directory '/home/nrb/go/src/github.com/vmware-tanzu/velero'
cd hack/build-image && docker build --pull -t velero-builder .
Sending build context to Docker daemon  3.584kB
Step 1/26 : FROM golang:1.12
1.12: Pulling from library/golang
Digest: sha256:53b838583589af3bddf26c5217343acac84313751d8655e666cc1309279134e2
Status: Image is up to date for golang:1.12
 ---> bc0268f5ce47
Step 2/26 : RUN mkdir -p /go/src/k8s.io
 ---> Using cache
 ---> da630f7acb4f
Step 3/26 : WORKDIR /go/src/k8s.io
 ---> Using cache
 ---> 101016df042b
Step 4/26 : RUN git config --global advice.detachedHead false
 ---> Using cache
 ---> 9f719c2c2fc8
Step 5/26 : RUN git clone -b kubernetes-1.15.3 https://github.com/kubernetes/code-generator
 ---> Using cache
 ---> 480f47055161
Step 6/26 : RUN git clone -b kubernetes-1.15.3 https://github.com/kubernetes/apimachinery
 ---> Using cache
 ---> dddf0db49eea
Step 7/26 : WORKDIR  /go/src/k8s.io/code-generator
 ---> Using cache
 ---> c47589d55bb2
Step 8/26 : RUN GO111MODULE=on go mod vendor
 ---> Running in 9037a0493089
go: finding github.com/spf13/pflag v1.0.1
go: finding github.com/gogo/protobuf v0.0.0-20171007142547-342cbe0a0415
go: finding k8s.io/gengo v0.0.0-20190116091435-f8a0810f38af
go: finding k8s.io/klog v0.3.1
go: finding golang.org/x/tools v0.0.0-20190313210603-aa82965741a9
go: gonum.org/v1/gonum@v0.0.0-20190331200053-3d26580ed485: unrecognized import path "gonum.org/v1/gonum" (https fetch: Get https://gonum.org/v1/gonum?go-get=1: dial tcp 104.248.78.23:443: i/o timeout)
go: gonum.org/v1/netlib@v0.0.0-20190331212654-76723241ea4e: unrecognized import path "gonum.org/v1/netlib" (https fetch: Get https://gonum.org/v1/netlib?go-get=1: dial tcp 104.248.78.23:443: i/o timeout)
go: error loading module requirements
The command '/bin/sh -c GO111MODULE=on go mod vendor' returned a non-zero code: 1
make[2]: *** [Makefile:206: build-image] Error 1
make[2]: Leaving directory '/home/nrb/go/src/github.com/vmware-tanzu/velero'
make[1]: *** [Makefile:99: _output/bin/linux/amd64/velero] Error 2
make[1]: Leaving directory '/home/nrb/go/src/github.com/vmware-tanzu/velero'
make: *** [Makefile:68: all] Error 2
```


**What did you expect to happen:**
 
`make` should be able to build regardless of the gonum.org availbility.

**Anything else you would like to add:**

This is blocking CI builds as well.

--- a/hack/build-image/Dockerfile
+++ b/hack/build-image/Dockerfile
@@ -19,6 +19,8 @@ WORKDIR /go/src/k8s.io
 RUN git config --global advice.detachedHead false
 RUN git clone -b kubernetes-1.15.3 https://github.com/kubernetes/code-generator
 RUN git clone -b kubernetes-1.15.3 https://github.com/kubernetes/apimachinery 
+# Use a proxy for go modules to reduce the likelihood of various hosts being down and breaking the build
+ENV GOPROXY=https://proxy.golang.org
 # vendor code-generator go modules to be compatible with pre-1.15
 WORKDIR  /go/src/k8s.io/code-generator
 # Don't use ENV here because we don't want to disable modules for subsequent commands

